xwiki-commons-core/xwiki-commons-xml/src/main/java/org/xwiki/xml/XMLUtils.java

```diff
@@ -225,6 +225,7 @@ public static String unescapeXMLComment(String content)
      * @param content the text to escape, may be {@code null}
      * @return a new escaped {@code String}, {@code null} if {@code null} input
      */
+    @Deprecated
```

Comment:
As I mentioned on some chat, I'm not sure we wish to deprecate this. Note that whenever a deprecation is done it also requires a `@deprecated` javadoc tag with explanation as to what should be used and in which version it's been deprecated.

Reply:
There is no reason to deprecate this.

Reply:
Well there's a reason :) It's very unprecise/misleading and leads to extra escapes that are not needed in some situations. I'm fine to keep it but the javadoc needs to be improved to explain that (the extra escapes) and we need to recommended using the other 2 methods whenever possible.

Reply:
Deprecated annotation removed and javadoc updated here https://github.com/xwiki/xwiki-commons/pull/102/files#diff-c1eed56adbad8fd67a8e196907ce032fR221
- Some more explanations of why { is escaped
- Explanations of why is is best to use `escapeAttributeValue` and `escapeElementText` instead of `escape`